src/ucp/wireup/wireup.c

```diff
@@ -46,10 +46,13 @@
 
 size_t ucp_wireup_msg_pack(void *dest, void *arg)
 {
-    ucp_request_t *req = arg;
-    *(ucp_wireup_msg_t*)dest = req->send.wireup;
-    memcpy((ucp_wireup_msg_t*)dest + 1, req->send.buffer, req->send.length);
-    return sizeof(ucp_wireup_msg_t) + req->send.length;
+    struct iovec *wireup_msg_iov = (struct iovec*)arg;
+
+    memcpy(dest, wireup_msg_iov[0].iov_base, wireup_msg_iov[0].iov_len);
+    memcpy(UCS_PTR_BYTE_OFFSET(dest, wireup_msg_iov[0].iov_len),
+           wireup_msg_iov[1].iov_base, wireup_msg_iov[1].iov_len);
```

Comment:
minor: is it possible to reuse some iov copy function (but only if code will be shorter)?

Reply:
done